app/src/main/java/org/oppia/app/player/exploration/ExplorationActivity.kt

```diff
@@ -29,4 +32,25 @@ class ExplorationActivity : InjectableAppCompatActivity() {
       return intent
     }
   }
+
+  override fun onBackPressed() {
+    showStopExplorationDialogFragment()
+  }
+
+  private fun showStopExplorationDialogFragment() {
+    val previousFragment = supportFragmentManager.findFragmentByTag(TAG_STOP_EXPLORATION_DIALOG)
+    if (previousFragment != null) {
+      supportFragmentManager.beginTransaction().remove(previousFragment).commitNow()
+    }
+    val dialogFragment = StopExplorationDialogFragment.newInstance()
+    dialogFragment.showNow(supportFragmentManager, TAG_STOP_EXPLORATION_DIALOG)
+  }
+
+  override fun stopExploration() {
+    explorationActivityPresenter.stopExploration()
+  }
+
+  fun routeBackToTopicActivity(){
```

Comment:
Nit: space before {

Reply:
Removed this function because this is no more needed after making change on above comment.